From 7807c44a824b7e19d3ee341567796cafdc594bff Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Wed, 7 Sep 2022 05:39:58 +0000
Subject: [PATCH] resolve merge conflicts of
 9c93920de7a8d8106a1bc2e11ebfed79df66946b to sc-dev

Ignore-AOSP-First: Security
Test: I solemnly swear I tested this conflict resolution.
Bug: 245420598
Change-Id: I029b5e310485e0180e4b7c7280db022848fcd646
(cherry picked from commit be2f542a6a1fb7f89b4562bc149cfdb8ac02748d)
Merged-In: I029b5e310485e0180e4b7c7280db022848fcd646
---
 btif/src/bluetooth.cc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/btif/src/bluetooth.cc b/btif/src/bluetooth.cc
index a72f787f5..e684c1235 100644
--- a/btif/src/bluetooth.cc
+++ b/btif/src/bluetooth.cc
@@ -359,11 +359,14 @@ static int get_connection_state(const RawAddress* bd_addr) {
 
 static int pin_reply(const RawAddress* bd_addr, uint8_t accept, uint8_t pin_len,
                      bt_pin_code_t* pin_code) {
+  bt_pin_code_t tmp_pin_code;
+  /* sanity check */
   if (!interface_ready()) return BT_STATUS_NOT_READY;
   if (pin_code == nullptr || pin_len > PIN_CODE_LEN) return BT_STATUS_FAIL;
 
+  memcpy(&tmp_pin_code, pin_code, pin_len);
   do_in_main_thread(FROM_HERE, base::BindOnce(btif_dm_pin_reply, *bd_addr,
-                                              accept, pin_len, *pin_code));
+                                              accept, pin_len, tmp_pin_code));
   return BT_STATUS_SUCCESS;
 }
 
-- 
2.38.0.413.g74048e4d9e-goog

